staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -500,24 +541,47 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		doUnconditionalUpdate := newResourceVersion == 0 && e.UpdateStrategy.AllowUnconditionalUpdate()
 
 		if existingResourceVersion == 0 {
-			creating = true
-			creatingObj = obj
-			if err := rest.BeforeCreate(e.CreateStrategy, ctx, obj); err != nil {
-				return nil, nil, err
+			var cleanup func(runtime.Object, error)
```

Comment:
I haven't looked carefully yet, but this means we could call BeginCreate multiple times...

Reply:
> I haven't looked carefully yet, but this means we could call BeginCreate multiple times...

If I'm reading correctly, the cleanup would still be called, so if the caller built his cleanupFn correctly, it should.

Reply:
Yes (but not without cleanup in between).  The whole logic here is called from within GuaranteedUpdate() so (if I understand) it's possible but unlikely that create could fail and get retried already.  If we don't want that behavior we should factor that out of the func() in there.